src/hooks/useInput.tsx

```diff
@@ -0,0 +1,13 @@
+import { useState, ChangeEvent } from 'react';
+
+const useInput = (initialState: string) => {
+    const [value, setValue] = useState(initialState);
+
+    const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setValue(e.target.value);
+    };
+
+    return { value, setValue, onChange };
+};
```

Comment:
제 코드에서 많은 부분을 참고하셨으니 추가로 더 말씀을 드려보자면, 제가 일부러 상태와 메서드의 이름을 `value`, `onChange`로 지은 이유는, 컴포넌트에서 바인딩할 때 코드를 더 줄일 수 있기 때문입니다.

```js
import useInput from 'hooks'

...

const bindInput = useInput('')
return
...
  <input {...bindInput} />
...
```
이런식으로 코드를 크게 줄일수 있죠..! 하지만 setValue는 useForm을 위해 같이 리턴시킨 것이기 때문에, 해당 함수를
```suggestion
const useInput = (initialState: string) => {
    const [value, setValue] = useState(initialState);

    const onChange = (e: ChangeEvent<HTMLInputElement>) => {
        setValue(e.target.value);
    };

    return { inputBindings: { value, onChange}, setValue };
};
```
로 바꾼다면 `<input/>` 만 사용하는 컴포넌트에서도 `useInput`을 사용할 수 있겠죠?

Reply:
대박이네요... 이게 진짜 컴포넌트 같습니다...